lib/spack/spack/spec.py

```diff
@@ -3257,8 +3250,14 @@ def spec(self, name):
             spec_namespace = None
             spec_name = None
 
-        # This will init the spec without calling __init__.
-        spec = Spec.__new__(Spec)
+        if self._initial is None:
+            # This will init the spec without calling Spec.__init__
+            spec = Spec.__new__(Spec)
+        else:
+            # this is used by Spec.__init__
+            spec = self._initial
+            self._initial = None
```

Comment:
Minor point. I am wondering if writing this like:
```python
spec, self._initial = self._initial, None
```
will make even more evident that we are consuming `self._initial`.

Reply:
I had to read that twice to make sure nothing tricky was going on, and I had to think about multiple assignment evaluation order in Python, so maybe not? 😄